js_src/modal.ts

```diff
@@ -0,0 +1,51 @@
+class Modal {
+    public readonly isPositionedAbsolute;
+    public readonly isDanger;
+
+    constructor(public element) {
+        this.isPositionedAbsolute = this.element.classList.contains("is-with-backdrop");
+        this.isDanger = this.element.classList.contains("is-danger");
+    }
+
+    private toggle() {
+        if (!this.isPositionedAbsolute) {
+            this.element.style.top = document.scrollingElement.scrollTop + "px";
+        }
+
+        this.element.classList.toggle("is-active");
+    }
+
+    static find(query) {
+        const elements = document.querySelectorAll(query);
+        let toggles = [];
+        for (let i = 0; i < elements.length; i++) {
```

Comment:
Comments below about using `.map` apply here too - also look for other places it might be applicable.

Reply:
see the comments on the other reviews/discussion in #front-end-development in Discord.